### ISSUE TITLE
Required fields validation error leads to "Entry draft not found" when saving draft for review

### ISSUE THREAD

**Issue by riusi88** [engram-design/Workflow#148]
**Description**
An error saving a draft occurs when there are required fields in the entry that are left empty, and an Editor attempts to "Save draft and submit for review". If they submit a draft for review without filling out the required fields, the user is presented with the familiar Field cannot be blank" validation error. However, after this point when the Editor fills out the required fields and saves the draft for review, they receive an "Entry draft not found" error. 

I imagine this has something to do with the DraftID not being set properly when the page refreshes and displays the required field validation errors. 

**Steps to reproduce**
1. Enable Workflow for a Section that includes required fields
2. As an Editor, edit an Entry within the Section above, save draft for review without entering data into a required field
3. Page refreshes with required field validation errors. "Field ___ cannot be blank"
4. Enter data into required fields and save draft for review
5. Receive "Entry draft not found" error

**Additional info**

- Plugin version: dev-craft-3 as 1.7.2
- I submitted this issue https://github.com/verbb/workflow/issues/145 so have this version of the plugin currently installed.
- Craft version: Craft Pro 3.7.37
- Multi-site: No

**Error Report**
`yii\web\NotFoundHttpException: Entry draft not found in /Users/user/MAMP Hosts/website/craft3/vendor/craftcms/cms/src/controllers/EntryRevisionsController.php:252
Stack trace:
#0 [internal function]: craft\controllers\EntryRevisionsController->actionSaveDraft()
#1 /Users/user/MAMP Hosts/website/craft3/vendor/yiisoft/yii2/base/InlineAction.php(57): call_user_func_array(Array, Array)
#2 /Users/user/MAMP Hosts/website/craft3/vendor/yiisoft/yii2/base/Controller.php(178): yii\base\InlineAction->runWithParams(Array)
#3 /Users/user/MAMP Hosts/website/craft3/vendor/yiisoft/yii2/base/Module.php(552): yii\base\Controller->runAction('save-draft', Array)
#4 /Users/user/MAMP Hosts/website/craft3/vendor/craftcms/cms/src/web/Application.php(293): yii\base\Module->runAction('entry-revisions...', Array)
#5 /Users/user/MAMP Hosts/website/craft3/vendor/verbb/workflow/src/controllers/SubmissionsController.php(95): craft\web\Application->runAction('entry-revisions...')
#6 [internal function]: verbb\workflow\controllers\SubmissionsController->actionSaveDraft()
#7 /Users/user/MAMP Hosts/website/craft3/vendor/yiisoft/yii2/base/InlineAction.php(57): call_user_func_array(Array, Array)
#8 /Users/user/MAMP Hosts/website/craft3/vendor/yiisoft/yii2/base/Controller.php(178): yii\base\InlineAction->runWithParams(Array)
#9 /Users/user/MAMP Hosts/website/craft3/vendor/yiisoft/yii2/base/Module.php(552): yii\base\Controller->runAction('save-draft', Array)
#10 /Users/user/MAMP Hosts/website/craft3/vendor/craftcms/cms/src/web/Application.php(293): yii\base\Module->runAction('workflow/submis...', Array)
#11 /Users/user/MAMP Hosts/website/craft3/vendor/craftcms/cms/src/web/Application.php(602): craft\web\Application->runAction('workflow/submis...', Array)
#12 /Users/user/MAMP Hosts/website/craft3/vendor/craftcms/cms/src/web/Application.php(272): craft\web\Application->_processActionRequest(Object(craft\web\Request))
#13 /Users/user/MAMP Hosts/website/craft3/vendor/yiisoft/yii2/base/Application.php(384): craft\web\Application->handleRequest(Object(craft\web\Request))
#14 /Users/user/MAMP Hosts/website/craft3/web/index.php(12): yii\base\Application->run()
#15 {main}`

**Comment by engram-design** [comment]
Looks like this only happens when creating a new draft from an **existing** entry. Fixed for the next release. To get the fix early, change your `verbb/workflow` requirement in `composer.json` to:

```
"require": {
  "verbb/workflow": "dev-craft-3 as 1.7.2",
  "...": "..."
}
```

Then run `composer update`.

**Comment by riusi88** [comment]
Thank you for the fix! Once again I appreciate the quick turnaround on this.

**Comment by riusi88** [comment]
hey @engram-design, ran into a few bugs when installing the update. 

Following the steps to tigger the initial issue:
- After clicking "save draft for review", it refreshes the page into the "current" entry, and not the Draft that should have been created.
- The Draft that the Editor attempted to make in step 1 doesn't appear in the Draft/Revision dropdown
- The notification sends to the Reviewer (with proper notes), but the direct link to the Draft doesn't work due to there being no Draft with that ID

It appears like the Draft isn't saved, but the notifications send out as expected.

**Comment by engram-design** [comment]
Alright, I think my previous fix wasn't really addressing the root cause of the issue. I believe I've gotten to the bottom of it, where if it's failing validation, it'll need to set `isProvisionalDraft = true` back again, due to how late in the entry-saving life cycle it appears.

The `entry-revisions/save-draft` controller will have already set this to `false` before our `Entry::EVENT_BEFORE_SAVE` event has run, so it needs to switch it back when failing, to continue editing the provisional draft.

Feel free to update again, hopefully for the last time.

**Comment by engram-design** [comment]
Fixed in [1.7.3](https://github.com/verbb/workflow/releases/tag/1.7.3)
